### DISCUSSION TITLE
button组件你当时是怎么设计的呢？

### DISCUSSION THREAD

**Discussion by keer-tea** [FightingDesign/fighting-design#256] [General]
1，我已经看了一点源码，font-size，font-color，size，disabled，type，text，simple，color这些属性，可能互相有影响。

比如组件的font-size和size属性都设置了按钮的`font-size`css属性，那到底该听谁的呢？

2，组件的props已经有默认值了，在使用css变量的时候又加上了默认值，感觉只保留其中一个默认值会好一点

3，有可能单独一个prop设置不会生效，需要有两个才会。比如设置了color属性后，设置font-color属性才会生效

单独设置font-color属性不会生效，这算不算一个bug？

4，还有按钮的文字水平垂直居中既然是用的flex来实现，那么我感觉line-height就可以注释了

5，文字按钮和简约按钮，区别感觉不是很大。。

**Comment by Tyh2001** [comment]
是的，可以 font-color，这确实是个 bug，按钮组件功能太多太杂了，是要优化下了

**Comment by keer-tea** [comment]
我先来试着改下吧

**Reply by Tyh2001** [reply]
> 我先来试着改下吧

没问题 👌👌
